synapse/types.py

```diff
@@ -920,3 +921,22 @@ class UserProfile(TypedDict):
 class RetentionPolicy:
     min_lifetime: Optional[int] = None
     max_lifetime: Optional[int] = None
+
+
+class SynapseBaseModel(BaseModel):
```

Comment:
FWIW I'd naively expect that a name like `synapse.types` ought to be something that you can import everywhere.

I considered putting in `rest`. But we might also want to use this for parsing the config. Though having said that I'm not sure if "ignore unknown fields" makes sense for config.

Happy to move it to `rest` now rather than anticipating the future.